File: src/main/kotlin/fr/unilim/saes5/model/WordAnalytics.java
package fr.unilim.saes5.model;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class WordAnalytics {

    public boolean isWordPresent(List<Word> words) {
        return words != null && !words.isEmpty();
    }

    public float wordRatio(Word word, List<Word> words) {
        if (words == null || words.isEmpty()) {
            return 0.0f;
        }

        long count = words.stream().filter(w -> w.equals(word)).count();
        return ((float) count / words.size()) * 100;
    }

    public Map<Word, Integer> wordRank(List<Word> words) {
        Map<Word, Integer> wordCount = new HashMap<>();

        if (words != null) {
            for (Word word : words) {
                wordCount.put(word, wordCount.getOrDefault(word, 0) + 1);
            }
        }

        return wordCount;
    }
}

File: src/main/kotlin/fr/unilim/saes5/model/WordAnalytics.kt
package fr.unilim.saes5.model

class WordAnalytics {
    fun isWordPresent(words: List<Word?>?, word: Word?): Boolean {
        return words!!.contains(word);
    }

    fun wordRatio(word: Word?, words: List<Word>?): Float {
        if (words.isNullOrEmpty()) {
            return 0.0f
        }
        val count = words.stream().filter { w: Word -> w.equals(word) }.count()
        return count.toFloat() / words.size * 100
    }

    fun wordRank(words: List<Word>?): Map<Word, Int> {
        val wordCount: MutableMap<Word, Int> = HashMap()
        if (words != null) {
            for (word in words) {
                wordCount[word] = wordCount.getOrDefault(word, 0) + 1
            }
        }
        return wordCount
    }
}
